compiler+runtime/test/jank/cpp/try/pass-catch-cpp-std-exception.jank

```diff
@@ -0,0 +1,11 @@
+;; Test catching std::exception base class
+(cpp/raw "void throw_runtime_error_as_exception() { throw std::runtime_error(\"derived error\"); }")
+
+(let [caught (atom false)]
+  (try
+    (cpp/throw_runtime_error_as_exception)
+    (catch cpp/std.exception e
+      (reset! caught true)))
```

Comment:
It'd be great to grab `(cpp/.what e)` instead of just `true` here. That way we can verify the exception is valid. I know from experience, with LLVM IR, that it's too easy for things to look like they're working when they're not. That's why our tests should check actual values, too.